Wrap OpenRA.exe arguments in quotes

Xinayder opened the pull request:
On Windows, if the path to the replay contains spaces, it wouldn't be parsed correctly and would crash the game.

Fixes #10572

RoosterDragon commented:
This won't be sufficient as the arguments are not passed quoted from the shell command: [OpenRA.nsi#L62](https://github.com/OpenRA/OpenRA/blob/ce8b03a276f2684f8df3dae33aa5849fba211e4b/packaging/windows/OpenRA.nsi#L62)

You'll need to add some quotes there too, otherwise your efforts to preserve the args here are for nothing because the args will already have been split. If you do that, then this mostly works.

Xinayder committed:
Wrap OpenRA arguments in quotes

On Windows, if the path to the replay contained spaces, it wouldn't be parsed correctly and would crash the game.

RoosterDragon commented on packaging/windows/OpenRA.nsi:
```diff
@@ -59,7 +59,7 @@ Section "-Reg" Reg
 	WriteRegStr HKLM "Software\Classes\.orarep" "" "OpenRA_replay"
 	WriteRegStr HKLM "Software\Classes\OpenRA_replay\DefaultIcon" "" "$INSTDIR\OpenRA.ico,0"
-	WriteRegStr HKLM "Software\Classes\OpenRA_replay\Shell\Open\Command" "" "$INSTDIR\OpenRA.exe Launch.Replay=%1"
+	WriteRegStr HKLM "Software\Classes\OpenRA_replay\Shell\Open\Command" "" "$INSTDIR\OpenRA.exe Launch.Replay='%1'"
```
Single quotes don't work.

RockyTV replied:
@RoosterDragon Weird. I just tested again and it still works by passing the replay path with single quotes. How are you testing it?

RoosterDragon replied:
I was running this via the debugger because I'm lazy - perhaps the single quotes act differently when launching a replay properly. I'll remove my changes requested for the moment.

RoosterDragon replied:
Nope definitely doesn't work.
```
Exception of type `System.NotSupportedException`: The given path's format is not supported.
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy)
   at System.IO.FileStream..ctor(String path, FileMode mode)
   at OpenRA.FileFormats.ReplayMetadata.Read(String path)
   at OpenRA.Mods.Common.LoadScreens.BlankLoadScreen.StartGame(Arguments args)
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Game.Initialize(Arguments args)
   at OpenRA.Program.Run(String[] args)
   at OpenRA.Program.Main(String[] args)
```

If I edit the registry key thusly, things work:
`...\OpenRA.exe "Launch.Replay=%1"`

RockyTV commented:
Tested here, and it works now.

RoosterDragon commented:
:+1: Works for me on Windows.

Could somebody on not-Windows confirm this doesn't break their ability to launch replays?

pchote commented:
This is all windows-only code, so it won't affect other platforms.

abcdefg30 commented:
Weird, it still doesn't work for me when I use
`OpenRA.exe Launch.Replay=C:\Users\User\Downloads\test test\testreplay.orarep` or
`OpenRA.exe Launch.Replay=C:\Users\User\Downloads\test replay.orarep`

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/fb6087d97e29e46a8e11e530a616e953a56e2fcf...ba17583c3e6eb37e5fe55cc952a0c35c3ebd941f)
